src/main/scala/is/hail/io/vcf/HtsjdkRecordReader.scala

```diff
@@ -191,18 +191,21 @@ object HtsjdkRecordReader {
   private val haploidRegex = """^[0-9]+$""".r
   private val diploidRegex = """^([0-9]+)[|/]([0-9]+)$""".r
 
-  def parseCall(gt: String, nAlleles: Int): Call = {
-    val call: Call = gt match {
-      case diploidRegex(a0, a1) => Call(Genotype.gtIndexWithSwap(a0.toInt, a1.toInt))
+  def parseCall(gt: String, nAlleles: Int): BoxedCall = {
+    gt match {
+      case diploidRegex(a0, a1) =>
```

Comment:
I considered passing the region value builder directly so as to avoid boxing. However, some of the tests relied on this function and it would be difficult to alter the test with `rvb`. If it will make a huge performance difference, I'll switch it to the unboxed version.